pkg/util/os.go

```diff
@@ -1,16 +1,71 @@
 package util
 
 import (
+	"io"
 	"os"
 )
 
-// Create a directory including its parents.  Returns nil if directory already exists.
-func Mkdir(dir string) error {
-	if _, err := os.Stat(dir); os.IsNotExist(err) {
-		err = os.MkdirAll(dir, os.ModePerm)
+// Create a file if it does not exist.  Returns nil if 'file' already exists.
+func Create(file string) error {
+	if _, err := os.Stat(file); os.IsNotExist(err) {
+		file, err := os.Create(file)
 		if err != nil {
 			return err
 		}
+		defer file.Close()
+	}
+
+	return nil
+}
+
+// Delete a file if it exists.  Returns nil if 'file' does not exist.
+func Delete(file string) error {
+	var err error
+	if err = os.Remove(file); os.IsNotExist(err) {
+		return nil
+	}
+
+	return err
+}
+
+// Create a directory 'dir' including its parents.  Returns nil if directory already exists.
+func Mkdir(dir string) error {
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
+		return os.MkdirAll(dir, os.ModePerm)
+	}
+
+	return nil
+}
```

Comment:
does `os.MkdirAll` fail if (part of) the path given as argument already exists? It should not: https://go.dev/play/p/9jO8toFtYgf (hence `util.Mkdir` seems redundant?)

Reply:
Correct: "If path is already a directory, MkdirAll does nothing and returns nil. "

Reply:
Good point, I'll remove the function.